### PR TITLE
Ria 3211 rj ftpa decided

### PR DESCRIPTION
**Pull request by jkumar-moj**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RIA-3211

### Change description ###
Aggregated flags to manage the event pages in ftpa decided state

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaDecisionMidEvent.java [hunk 1]
```diff
@@ -47,7 +49,7 @@ public boolean canHandle(
             .orElseThrow(() -> new IllegalStateException("FtpaApplicantType is not present"));
 
         Optional<String> ftpaSubmitted = asylumCase.read(
-            ftpaApplicantType.equals("appellant") == true ? FTPA_APPELLANT_SUBMITTED : FTPA_RESPONDENT_SUBMITTED, String.class);
+            ftpaApplicantType.equals("appellant") ? FTPA_APPELLANT_SUBMITTED : FTPA_RESPONDENT_SUBMITTED, String.class);
```
Comment (Lee365) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaDecisionMidEvent.java:53]: "appellant" - Shouldn't this be a constant ?

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaDecisionMidEventTest.java [hunk 2]
```diff
@@ -169,4 +167,447 @@ public void should_successfully_record_ftpa_respondent_decision() {
...
+        final Set<String> errors = callbackResponse.getErrors();
+        assertThat(errors).isEmpty();
+        verify(asylumCase).write(FTPA_APPELLANT_NOTICE_OF_DECISION_SET_ASIDE_VISIBLE, YesOrNo.YES);
+    }
+
+    @Test
+    public void should_successfully_set_appellant_notice_of_decision_set_aside_visibility_when_partially_granted() {
+
+        when(callback.getEvent()).thenReturn(Event.RESIDENT_JUDGE_FTPA_DECISION);
+        when(asylumCase.read(FTPA_APPLICANT_TYPE, String.class)).thenReturn(Optional.of("appellant"));
```
Comment (Lee365) [inline src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaDecisionMidEventTest.java:275]: "appellant" - Shouldn't this be a constant taken from FTPA_APPLICANT_TYPE ?

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaDecisionMidEventTest.java [hunk 3]
```diff
@@ -169,4 +167,447 @@ public void should_successfully_record_ftpa_respondent_decision() {
...
+        when(asylumCase.read(FTPA_APPELLANT_SUBMITTED, String.class)).thenReturn(Optional.of("YES"));
```
Comment (Lee365) [inline src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaDecisionMidEventTest.java:276]: Shouldn't this be YesOrNo.YES ?

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaDecisionMidEventTest.java [hunk 4]
```diff
@@ -169,4 +167,447 @@ public void should_successfully_record_ftpa_respondent_decision() {
...
+
+        when(asylumCase.read(FTPA_APPELLANT_RJ_DECISION_OUTCOME_TYPE, String.class)).thenReturn(Optional.of("partiallyGranted"));
```
Comment (Lee365) [inline src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaDecisionMidEventTest.java:278]: Shouldn't partiallyGranted be taken from constant ?

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaDecisionMidEventTest.java [hunk 5]
```diff
@@ -169,4 +167,447 @@ public void should_successfully_record_ftpa_respondent_decision() {
         final Set<String> errors = callbackResponse.getErrors();
         assertThat(errors).isEmpty();
     }
+
+    @Test
+    public void should_successfully_set_ftpa_appellant_decision_reasons_notes_visibility_when_granted() {
+
+        when(callback.getEvent()).thenReturn(Event.LEADERSHIP_JUDGE_FTPA_DECISION);
+        when(asylumCase.read(FTPA_APPLICANT_TYPE, String.class)).thenReturn(Optional.of("appellant"));
```
Comment (Lee365) [inline src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaDecisionMidEventTest.java:175]: "appellant" - Shouldn't this be a constant ?

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaDecisionMidEventTest.java [hunk 6]
```diff
@@ -169,4 +167,447 @@ public void should_successfully_record_ftpa_respondent_decision() {
...
+        assertThat(errors).isEmpty();
+        verify(asylumCase).write(FTPA_APPELLANT_NOTICE_OF_DECISION_SET_ASIDE_VISIBLE, YesOrNo.YES);
+    }
+
+    @Test
+    public void should_successfully_set_appellant_notice_of_decision_set_aside_visibility_when_refused() {
+
+        when(callback.getEvent()).thenReturn(Event.RESIDENT_JUDGE_FTPA_DECISION);
+        when(asylumCase.read(FTPA_APPLICANT_TYPE, String.class)).thenReturn(Optional.of("appellant"));
+        when(asylumCase.read(FTPA_APPELLANT_SUBMITTED, String.class)).thenReturn(Optional.of("YES"));
```
Comment (Lee365) [inline src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaDecisionMidEventTest.java:296]: Shouldn't this be YesOrNo.YES ?

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaDecisionMidEventTest.java [hunk 7]
```diff
@@ -169,4 +167,447 @@ public void should_successfully_record_ftpa_respondent_decision() {
...
+        final Set<String> errors = callbackResponse.getErrors();
+        assertThat(errors).isEmpty();
+        verify(asylumCase).write(FTPA_APPELLANT_NOTICE_OF_DECISION_SET_ASIDE_VISIBLE, YesOrNo.YES);
+    }
+
+    @Test
+    public void should_successfully_set_appellant_notice_of_decision_set_aside_visibility_when_refused() {
+
+        when(callback.getEvent()).thenReturn(Event.RESIDENT_JUDGE_FTPA_DECISION);
+        when(asylumCase.read(FTPA_APPLICANT_TYPE, String.class)).thenReturn(Optional.of("appellant"));
```
Comment (Lee365) [inline src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaDecisionMidEventTest.java:295]: "appellant" - Shouldn't this be a constant taken from FTPA_APPLICANT_TYPE ?

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaDecisionMidEventTest.java [hunk 8]
```diff
@@ -169,4 +167,447 @@ public void should_successfully_record_ftpa_respondent_decision() {
...
+        when(asylumCase.read(FTPA_APPELLANT_SUBMITTED, String.class)).thenReturn(Optional.of("YES"));
+
+        when(asylumCase.read(FTPA_APPELLANT_RJ_DECISION_OUTCOME_TYPE, String.class)).thenReturn(Optional.of("refused"));
```
Comment (Lee365) [inline src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaDecisionMidEventTest.java:298]: "refused" - Shouldn't this be a constant  ?

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaDecisionMidEventTest.java [hunk 9]
```diff
@@ -169,4 +167,447 @@ public void should_successfully_record_ftpa_respondent_decision() {
...
+        final Set<String> errors = callbackResponse.getErrors();
+        assertThat(errors).isEmpty();
+        verify(asylumCase).write(FTPA_APPELLANT_DECISION_OBJECTIONS_VISIBLE, YesOrNo.YES);
+    }
+
+    @Test
+    public void should_successfully_set_respondent_decision_objections_page_visibility_when_set_aside_is_yes() {
+
+        when(callback.getEvent()).thenReturn(Event.LEADERSHIP_JUDGE_FTPA_DECISION);
+        when(asylumCase.read(FTPA_APPLICANT_TYPE, String.class)).thenReturn(Optional.of("respondent"));
```
Comment (Lee365) [inline src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaDecisionMidEventTest.java:396]: Not sure - why constants are not used ! Tests can also use it .

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaDecisionMidEvent.java (hunk 1 above)
Comment (jkumar-moj) [reply src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaDecisionMidEvent.java:53]: It is again defined as String type and there is no constant defined.

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaDecisionMidEventTest.java (hunk 2 above)
Comment (jkumar-moj) [reply src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaDecisionMidEventTest.java:275]: It is again defined as String type and there is no constant defined.

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaDecisionMidEventTest.java (hunk 3 above)
Comment (jkumar-moj) [reply src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaDecisionMidEventTest.java:276]: It is defined as a string type in AsylumFieldDefintions

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaDecisionMidEventTest.java (hunk 4 above)
Comment (jkumar-moj) [reply src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaDecisionMidEventTest.java:278]: Again it is defined as String type and read as string in the code.

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaDecisionMidEventTest.java (hunk 5 above)
Comment (jkumar-moj) [reply src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaDecisionMidEventTest.java:175]: same as above

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaDecisionMidEventTest.java (hunk 6 above)
Comment (jkumar-moj) [reply src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaDecisionMidEventTest.java:296]: Explained as above.

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaDecisionMidEventTest.java (hunk 7 above)
Comment (jkumar-moj) [reply src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaDecisionMidEventTest.java:295]: As explained above, FTPA_APPLICANT_TYPE is defined as a String type.

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaDecisionMidEventTest.java (hunk 8 above)
Comment (jkumar-moj) [reply src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaDecisionMidEventTest.java:298]: Similar to the above

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaDecisionMidEventTest.java (hunk 9 above)
Comment (jkumar-moj) [reply src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaDecisionMidEventTest.java:396]: For the reason that the fields are defined as String types

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaDecisionMidEvent.java (hunk 1 above)
Comment (Lee365) [reply src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FtpaDecisionMidEvent.java:53]: Sorry - can't it not be Parties.APPELLANT ?

**Pull request merged**